### PR TITLE
test: Prefer factory to create Integration objects in test setup

### PR DESCRIPTION
**Pull request by RyanSkonnord**
Introduce `create_provider_integration` as a test case factory method. Globally find and replace instances of `Integration.objects.create`. The purpose is to wrap them in `assume_test_silo_mode(SiloMode.CONTROL)` so that we can expand the number of test cases run in region mode by default.

Fix some incidental type-checking failures that the change exposed.

### CODE REVIEW

File: tests/sentry/api/endpoints/test_organization_repository_details.py [hunk 1]
```diff
@@ -235,7 +234,7 @@ def test_put_bad_integration_org(self):
 
         org = self.create_organization(owner=self.user, name="baz")
         with assume_test_silo_mode(SiloMode.CONTROL):
```
Comment (markstory) [inline tests/sentry/api/endpoints/test_organization_repository_details.py:236]: Should we clean up silo mode switches that are now redundant?

File: tests/sentry/api/endpoints/test_project_repo_path_parsing.py [hunk 2]
```diff
@@ -75,7 +74,7 @@ def test_wrong_file(self):
     def test_no_integration(self):
         # create the integration but don't install it
         with assume_test_silo_mode(SiloMode.CONTROL):
```
Comment (markstory) [inline tests/sentry/api/endpoints/test_project_repo_path_parsing.py:76]: This one looks redundant too.

**Review by markstory — APPROVED** [review]

**Review by dashed — APPROVED** [review]
🚀

**Comment by RyanSkonnord** [comment]
FYI, I accidentally committed [03f9cab](https://github.com/getsentry/sentry/pull/62874/commits/03f9cab9315b2e9871b09c57d974b9016cb5125f) to the wrong branch and thought it was in here yesterday. Feedback on naming choices welcome.

**Review by markstory — APPROVED** [review]

**Comment by RyanSkonnord** [comment]
> Should we clean up silo mode switches that are now redundant?

I got started on this and realized there were a lot of cases where we still need the mode switch a line down to call `add_organization`. I think it makes sense to fold that into another new factory method. That would require a rote sweep across a large number of cases, so I'll handle that in a separate branch.

**Pull request merged**